### PR TITLE
sdk protobuf support

### LINKED ISSUE #81: [GraphSDK] Define protobuf types for API inputs and Outputs
# Problem
The details of why we decided to go this route is mentioned in #77

# Acceptence criterea
- Define protobuf types for api input and outputs
- Generate Rust code for defined protobuf types
- Generate Java code for defined protobuf types

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is to add serialization and deserialization support between Java and Rust using protobuf

Closes #81 

# Discussion
- Added protobuf code generation support for Rust and java using `protoc`
- Created mapping for `initializeGraphState` api
- All Java files inside `java/lib/src/main/java/io/amplica/graphsdk/models/` are generated
- All Rust files inside `bridge/common/src/proto_types/` are generated

# Checklist
- [x] Java tests cover end to end serialization and deserialization

### CODE REVIEW

File: bridge/common/protos/input.proto [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    oneof inner {
+      ConnectAction connect_action = 1;
+      DisconnectAction disconnect_action = 2;
+      AddGraphKey add_key_action = 3;
+    }
+  }
+  repeated Action actions = 1;
+}
```
Comment (saraswatpuneet) [inline bridge/common/protos/input.proto:75]: are there some crates like https://docs.rs/autoproto/latest/autoproto/
that can automate this?

**Review by saraswatpuneet — APPROVED** [review]
looks good to me! nothing much to add just one comment

File: bridge/common/protos/input.proto (hunk 1 above)
Comment (aramikm) [reply bridge/common/protos/input.proto:75]: I think this is dependent on `prost` crate which we are not using and it says one of it's limitations is exactly what we are doing `nested oneof ` to support Rust style enums

**Comment by JoeCap08055** [comment]
Maybe we should integrate building the generated proto code into the build process, instead of checking the generated files in?

**Comment by aramikm** [comment]
I think we will eventually use `build.rs` to generate rust and java files when `protos` changed but for now to be able to run the tests we need those classes and structs to check-in.

**Pull request merged**